cylc/uiserver/resolvers.py

```
@@ -15,10 +15,189 @@
+
+    @staticmethod
+    async def play(workflows, args, workflows_mgr):
+        """Calls `cylc play`."""
+        response = []
+
+        # get ready to run the command
+        try:
+            # check that the requrest cylc version is available
+            cylc_version = None
+            if 'cylc_version' in args:
+                cylc_version = args['cylc_version']
+                if not check_cylc_version(cylc_version):
+                    return {'error': {'message': [
+                        f'cylc version not available: {cylc_version}'
+                    ]}}
+                args = dict(args)
+                args.pop('cylc_version')
+
+            # build the command
+            cmd = ['cylc', 'play', '--color=never']
+            for key, value in args.items():
+                if value is False:
+                    # don't add binary flags
+                    continue
+                key = snake_to_kebab(key)
+                if not isinstance(value, list):
+                    value = [value]
+                for item in value:
+                    cmd.append(quote(key))
+                    if item is not True:
+                        # don't provide values for binary flags
+                        if key == '--set':
+                            # TODO
+                            # don't quote this value as it messes with things
+                            # I think subprocess quotes everything anyway
+                            # double check
+                            cmd.append(item)
+                        else:
+                            cmd.append(quote(item))
+
+        except Exception as exc:
+            # oh noes, something went wrong, send back confirmation
+            response.append(
+                [
+                    {
+                        'error': {
+                            'message': str(exc)
+                        }
+                    }
+                ]
+            )
```

Should we return the error response here instead of continuing the execution flow?

---

Yeah, that would make sense!